raportit/myyntiseuranta.php

```diff
@@ -1105,19 +1105,19 @@ function vararvo($tuoteno, $vv, $kk, $pp) {
-            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,concat_ws('<br>',{$etuliite}.toim_nimi),concat_ws('<br>',{$etuliite}.nimi)){$ytgft} toim_nimi, ";
-            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,concat_ws('<br>',{$etuliite}.toim_nimitark),concat_ws('<br>',{$etuliite}.nimitark)){$ytgft} toim_nimitark, ";
-            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,{$etuliite}.toim_osoite,{$etuliite}.osoite){$ytgft} toim_osoite, ";
-            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,{$etuliite}.toim_postino,{$etuliite}.postino){$ytgft} toim_postino, ";
-            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,{$etuliite}.toim_postitp,{$etuliite}.postitp){$ytgft} toim_postitp, ";
-            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,{$etuliite}.toim_maa,{$etuliite}.maa){$ytgft} toim_maa, ";
+            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' ,concat_ws('<br>',{$etuliite}.toim_nimi),concat_ws('<br>',{$etuliite}.nimi)){$ytgft} toim_nimi, ";
+            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' ,concat_ws('<br>',{$etuliite}.toim_nimitark),concat_ws('<br>',{$etuliite}.nimitark)){$ytgft} toim_nimitark, ";
+            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' ,{$etuliite}.toim_osoite,{$etuliite}.osoite){$ytgft} toim_osoite, ";
+            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' ,{$etuliite}.toim_postino,{$etuliite}.postino){$ytgft} toim_postino, ";
+            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' ,{$etuliite}.toim_postitp,{$etuliite}.postitp){$ytgft} toim_postitp, ";
+            $select .= "{$ytgfe}if({$etuliite}.toim_nimi!='' ,{$etuliite}.toim_maa,{$etuliite}.maa){$ytgft} toim_maa, ";
             $select .= "{$ytgfe}if(asiakas.puhelin!='',asiakas.puhelin,asiakas.gsm){$ytgft} puhelin, ";
             $select .= "{$ytgfe}asiakas.email{$ytgft} email, ";
           }
           else {
             $select .= "{$etuliite}.ytunnus ytunnus, ";
             $select .= "{$ytgfe}{$etuliite}.toim_ovttunnus{$ytgft} toim_ovttunnus, ";
-            $select .= "{$ytgfe}concat_ws('<br>',concat_ws(' ',{$etuliite}.nimi,{$etuliite}.nimitark),if({$etuliite}.toim_nimi!='' and {$etuliite}.nimi!={$etuliite}.toim_nimi,concat_ws(' ',{$etuliite}.toim_nimi,{$etuliite}.toim_nimitark),NULL)){$ytgft} nimi, ";
+            $select .= "{$ytgfe}concat_ws('<br>',concat_ws(' ',{$etuliite}.nimi,{$etuliite}.nimitark),if({$etuliite}.toim_nimi!='' ,concat_ws(' ',{$etuliite}.toim_nimi,{$etuliite}.toim_nimitark),NULL)){$ytgft} nimi, ";
             $select .= "{$ytgfe}concat_ws('<br>',{$etuliite}.postitp,if({$etuliite}.toim_postitp!='' and {$etuliite}.postitp!={$etuliite}.toim_postitp,{$etuliite}.toim_postitp,NULL)){$ytgft} postitp, ";
```

Comment:
@satuaalto Tässä käy nyt niin, että jos nimi on sama ku toim_nimi, eli vain osoite on poikkeava, niin samat tiedot tulee rudulle kahteen kertaan. Eli tässä kuuluu olla toi "and {$etuliite}.nimi!={$etuliite}.toim_nimi" ehto. Mut tossa ylempänä tosiaan ei, koska siinä näytetään myös osoitetiedot.